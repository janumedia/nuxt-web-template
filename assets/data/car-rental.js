module.exports = {
    carsList:[
        {
            name: "Toyota Avanza",
            img: "images/contents/car-toyota-avanza.jpg",
            desc: "Price With Driver:<br><b>USD 90 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 5 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        },
        {
            name: "Suzuki APV",
            img: "images/contents/car-suzuki-apv.jpg",
            desc: "Price With Driver:<br><b>USD 85 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 5 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        },
        {
            name: "Suzuki Ertiga",
            img: "images/contents/car-suzuki-ertiga.jpg",
            desc: "Price With Driver:<br><b>USD 90 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 5 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        },
        {
            name: "Toyota Innova",
            img: "images/contents/car-toyota-innova.jpg",
            desc: "Price With Driver:<br><b>USD 130 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 5 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        },
        {
            name: "Kia Pregio",
            img: "images/contents/car-kia-pregio.jpg",
            desc: "Price With Driver:<br><b>USD 100 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 12 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        },
        {
            name: "Kia Travello",
            img: "images/contents/car-kia-travello.jpg",
            desc: "Price With Driver:<br><b>USD 100 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 12 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        },
        {
            name: "Isuzu Elf",
            img: "images/contents/car-isuzu-elf.jpg",
            desc: "Price With Driver:<br><b>USD 130 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 12 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        },
        {
            name: "Isuzu Elf Long",
            img: "images/contents/car-isuzu-elf-long.jpg",
            desc: "Price With Driver:<br><b>USD 130 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 12 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        },
        {
            name: "Toyota Hi Ace",
            img: "images/contents/car-toyota-hi-ace.jpg",
            desc: "Price With Driver:<br><b>USD 130 / 10 hours</b><br><br>Price Include:<br>Petrol<br>Air condition vehicles<br>Driver speak English<br>Capacity: passenger max 9 pax<br>Departure on Request",
            btnLabel: "Book Now",
            btnUrl: "book-now?type=booking-car"
        }
    ]
}