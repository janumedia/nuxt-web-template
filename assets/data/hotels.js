module.exports = {
    hotelsList:[
        {
            name: "Nusa Penida Beach Hotel",
            img: "images/contents/hotel-nusa-penida-beach-mid.jpg",
            desc: "Conveniently located in Bali, Nusa Penida Beach Hotel is a great base from which to explore this vibrant city. Situated only 20 km from the city center, guests are well located to enjoy the town's attractions and activities. With its convenient location, the hotel offers easy access to the city's must-see destinations.<br><br>Nusa Penida Beach Hotel also offers many facilities to enrich your stay in Bali. This hotel offers numerous on-site facilities to satisfy even the most discerning guest.",
            btnLabel: "Book Now",
            btnUrl: "/book-now?type=booking-tour"
        },
        {
            name: "Bintang Bungalows",
            img: "images/contents/hotel-bintang-bungalows-mid.jpg",
            desc: "Bintang Bungalow is located in Toyapakeh just about 800 meters from Crystal Bay. The hotel is a 15-minutes drive to Gamat Bay and 30-minutes to Angel's Billabong.<br><br>Features an outdoor pool. You can enjoy meals at the restaurant at the accommodation.",
            btnLabel: "Book Now",
            btnUrl: "/book-now?type=booking-tour"
        },
        {
            name: "Caspla Beach Hotel",
            img: "images/contents/hotel-caspla-beach-mid.jpg",
            desc: "Caspla Beach Hotel, Restaurant & Bar is built seashore side on the easthern most tip of Bali, that is Nusa Penida Island.<br><br>Enjoy luxury hotels on east Bali and an unforgettable holiday in Caspla Beach Hotel, Bar & Restaurant unique, elegant, tropical style, ocean front bungalows. the hotel building full of wood, here you can enjoy free pool, eating at the beachside restaurant.",
            btnLabel: "Book Now",
            btnUrl: "/book-now?type=booking-tour"
        },
        {
            name: "Semabu Hills Hotel",
            img: "images/contents/hotel-semabu-hills-mid.jpg",
            desc: "It can be argued that Bali is as close to heaven on earth as it gets and we at Semabu Hills have put in place the crowning glory of this picturesque getaway.<br><br>Just 30 minutes off the east coast of Bali on the tropical island of Nusa Penida, Semabu Hills is proud to host you and give you an experience to cherish.",
            btnLabel: "Book Now",
            btnUrl: "/book-now?type=booking-tour"
        }
    ]
}