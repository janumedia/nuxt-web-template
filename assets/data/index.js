module.exports = {
    destinationList: [
        {
            id:0,
            name: "One Day Tour Nusa Penida",
            img: "images/contents/crystal-bay-beach-mid-sq.jpg",
            desc: "One day tour exploring Nusa Penida:<br>Tree of Love and Klingking Beach,<br>Broken Beach and Angel Billabong,<br>Crystal Bay Beach<br><br><b>Start from Rp 400.000</b>",
            btnLabel: "More Details",
            url: "/tours/one-day-tour-nusa-penida"
        },
        {
            id: 1,
            name: "Snorkeling at Nusa Penida Beach",
            img: "images/contents/snorkeling-crystal-bay-beach-mid-sq.jpg",
            desc: "Enjoy beautiful underwater scene in Nusa Penida Beach<br>If you're lucky you may met Manta Ray or Mola-mola<br><br><b>Start from Rp 600.000</b>",
            btnLabel: "More Details",
            url: "/tours/one-day-tour-nusa-penida-plus-snorkeling"
        },
        {
            id: 2,
            name: "2 Days and 1 Night Tour Nusa Penida",
            img: "images/contents/raja-lima-sunset-mid-sq.jpg",
            desc: "Complete 2 days and 1 night tour covering west and east side of Nusa Penida famous tourist destionations<br><br><b>Start from Rp 750.000</b>",
            btnLabel: "more details",
            url: "/tours/2-days-1-night-tour-nusa-penida"
        },
        {
            id: 3,
            name: "One Day Tour Nusa Lembongan",
            desc: "One day tour within Nusa Lembongan:<br>Panorama Point, Dream Beach, Devil's Tears, Yellow Bridge and Mangrove Forest<br><br><b>Start from Rp 450.000</b>",
            img: "images/contents/devils-tears-mid-sq.jpg",
            btnLabel: "more details",
            url: "/tours/one-day-tour-nusa-lembongan"
        },
        {
            id: 4,
            name: "Nusa Penida Magical Place",
            desc: "You will guided to visiting Giri Putri Cave, a magical place inside the cave where you will discover a place of worship Hindu devotees and also Buddist and Confucian followers<br><br><b>Start from Rp 750.000</b>",
            img: "images/contents/giri-putri-cave-mid-sq.jpg",
            btnLabel: "more details",
            url: "/tours/2-days-1-night-tour-nusa-penida"
        }
        
    ]
}