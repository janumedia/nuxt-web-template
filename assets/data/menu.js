module.exports = {
    menu: [
        {
            name: "Home", path: "/", quote: "Cherysa Tour is a Bali and Nusa Penida Tour Agency"
        },
        {
            name: "Destinations", path: "/destinations", submenu: [
                { name: "Angel Billabong", path: "/destinations/angel-billabong"},
                { name: "Broken Beach", path: "/destinations/broken-beach"},
                { name: "Kelingking Beach", path: "/destinations/kelingking-beach"}
            ]
        },
        {
            name: "Hotels", path: "/hotels", quote: "Hotels in Nusa Penida Partnership with Cherysa Tour"
        },
        {
            name: "Contact", path: "/contact"
        },
        {
            name: "Sitemap", path: "/sitemap", mobileOnly: true
        }
    ]
}