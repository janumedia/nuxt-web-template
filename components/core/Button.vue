<template>
    <button :class="className" :type="submit !== undefined ? 'submit' : ''">
        <nuxt-link v-if="url && url.indexOf('http') == -1" :title="label" :to="url">{{ label }}</nuxt-link>
        <a v-else-if="url" :href="url" target="_blank" :title="label" rel="noopener">{{ label }}</a>
        <span v-else>{{ label }}</span>
    </button>
</template>
<script>
export default {
    props: ["primary", "submit", "label", "url"],
    computed: {
        className() {
            let str = this.primary !== undefined ? 'button--primary' : '';
            return str + (this.submit !== undefined ? ' button--submit' : '');
        }
    }
}
</script>
<style lang="scss" scoped>
@import "@/assets/css/base.scss";
button {
    border: 1px solid #ccc;;
    border-radius: 0;
    -webkit-border-radius: 0;
    -moz-border-radius: 0;    
    
    text-transform: uppercase;
    letter-spacing: 0.04em;
    display:inline-block;
    padding: 0;
    vertical-align:middle;
    text-decoration:none;
    font-size: 0.8em;
    color:#807e7e;
    background-color:inherit;
    text-align:center;
    cursor:pointer;
    white-space:nowrap;
    @include transition-duration(0.4s);

    &:focus {
        outline: none;
    }
    &:hover {
        color: $bgColor;
        background-color:#ccc;
    }
    & span,
    & a {
        display:block;
        color:#807e7e;
        padding:8px 16px;
        @include transition-duration(0.4s);
        cursor: pointer;
    }
    & > a:hover {
        color: $bgColor;
        background-color:#ccc;
    }
}

.button--primary {
    border: 1px solid $mainColor;
    color: $mainColor;
    padding:0px 9px;
    background-color: $mainColor;
    margin-bottom: 16px;
    span,
    a {
        color: $bgColor;
        &:hover {
            background-color:$errorColor;
        }
    }
    &:hover {
        background-color:$errorColor;
    }
}

.button--submit {
    padding:8px 40px;
}

@media only screen and (max-width:768px) {
    .button--submit {
        width: 100%;
    }
}

</style>
