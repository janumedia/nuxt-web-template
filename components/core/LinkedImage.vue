<template>
    <div :class="`${className} ${verticalAlign ? className + '--vertical-align' : ''}`">
        <nuxt-link v-if="url !== undefined && url.indexOf('http') == -1" :to="url" :title="title">
            <lazy-image 
                :src="src" 
                :title="title" 
                :square="square" 
                @onReady="imageReady" 
                @onLoaded="imageLoaded"
                @onError="imageError"
            />
            <div v-if="thumbnailMode" :class="`${className}__title`"><span>{{ title }}</span></div>
        </nuxt-link>
        <a v-else-if="url !== undefined && url.indexOf('http') > -1" :href="url" target="_blank" :title="title">
            <lazy-image 
                :src="src" 
                :title="title" 
                :square="square" 
                @onReady="imageReady" 
                @onLoaded="imageLoaded"
                @onError="imageError"
            />
        </a>
        <span v-else>
            <lazy-image 
                :src="src" 
                :title="title" 
                :square="square" 
                @onReady="imageReady" 
                @onLoaded="imageLoaded"
                @onError="imageError"
            />
        </span>
    </div>
</template>
<script>
import LazyImage from "@/components/core/LazyImage"
export default {
    props: ["src", "url", "title", "square", "thumbnailMode", "verticalAlign"],
    components: {
        LazyImage
    },
    data() {
        return {
            className: "linked-image"
        }
    },
    methods: {
        imageReady(el, index) {
            this.$emit('onReady', this, index);
        },
        imageLoaded(el, index) {
            this.$emit('onLoaded', this, index);
        },
        imageError(index) {
            this.$emit('onError', this, index);
        }
    }
}
</script>
<style lang="scss" scoped>
@import "@/assets/css/base.scss";

.linked-image {
    a {
        &:hover .linked-image__title {
            opacity: 1;
        }
    }
    &__title {
        text-align: center;
        color: rgba(255, 255, 255, 1);
        text-transform: uppercase;
        letter-spacing: 0.3em;
        font-weight: 700;
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        text-align: center;
        width:auto;
        background: rgba(0, 0, 0, 0.1);
        @include transition(opacity 0.35s);
        span {
            position: absolute;
            top: 50%;
            left: 0;
            right: 0;
            padding: 20px;
            @include translateY(-50%);
        }
    }
    &--vertical-align {
        position:absolute;
        top:0;
        left: 0;
        right: 0;
        width:auto;
        height: 100%;
        img {
            top:0;
        }
    }
}
// mouse hover supported only. usually for non touch devices //
@media only screen and (hover:hover) {
    .linked-image {
        a {
            div {
                opacity: 0.85;
                // fix move, scale issue //
                -webkit-filter: brightness(1.01);
                -moz-filter: brightness(1.01);
                -o-filter: brightness(1.01);
                -ms-filter: brightness(1.01);
                filter: brightness(1.01);
            }
            &:hover div {
                opacity: 1;
            }
        }
        &__title {
            opacity: 0;
        }
    }
}
</style>
