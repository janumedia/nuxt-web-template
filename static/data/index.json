{
    "title": "Nusa Penida Tour Guide",
    "desc": "Totalling some 200 square kilometres, Nusa Penida is much larger than the better known Nusa Lembongan. However, tourist infrastructure is very limited here but growing fast.\n\nDue to a lack of natural fresh water, little is grown or produced on Nusa Penida, and even some basic foodstuffs come in by boat. Visitors should therefore expect higher prices than in Bali, and not bank on any tourism-related luxury items being available for purchase here.",
    "metaImage": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/slides%2Fcrystal-bay-beach-2.jpg?alt=media&token=be8ffb8f-17e9-4b85-bf4a-9dff7f38348a",
    "body": [
        {
            "listBox": [
                {
                    "slides": [
                        {"name": "Peguyangan Water Spring", "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/slides%2Fpeguyangan-waterfall-3.jpg?alt=media&token=82e2b880-f354-468e-a857-f341a108e239", "credit": "@wanderersandwarriors"},
                        {"name": "Kelingking Beach", "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/slides%2Fklingking-beach.jpg?alt=media&token=0608592e-0bca-43ce-b535-857065b67ef5"},
                        {"name": "Angel Billabong", "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/slides%2Fangel-billabong.jpg?alt=media&token=9d4b2f99-3667-464e-9d6a-5c4b6850bf84"},
                        {"name": "Broken Beach", "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/slides%2Fbroken-beach.jpg?alt=media&token=d5c744e6-9fd4-4df0-8552-03659675f6d7"},
                        {"name": "Crystal Bay Beach", "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/slides%2Fcrystal-bay-beach-2.jpg?alt=media&token=be8ffb8f-17e9-4b85-bf4a-9dff7f38348a"}
                    ]
                }
            ]
        },
        {
            "name": "Recommendations",
            "chessGrid": [
                {
                    "name": "Angel Billabong",
                    "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/angel-billabong-mid-sq.jpg?alt=media&token=ef70e878-0de2-4b0c-bed8-513d3bc21758",
                    "desc": "a spectacular rock formation on Nusa Penida island's southwestern cliff edges. It's nearby Pasih Uug (Broken Beach), another popular site with unique limestone formations. The naturally formed rock lagoon offers a scenic seascape. You can descend into its crystal-clear rock pools for a swim or a soak",
                    "btnLabel": "View details",
                    "url": "/destinations/angel-billabong"
                },
                {
                    "name": "Broken Beach",
                    "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/broken-beach-mid-sq.jpg?alt=media&token=9769ac16-f00a-494d-afb1-709e96327d6e",
                    "desc": "Broken Beach also popular as Pasih Uug is a scenic coastal formation on the southwestern edge of Nusa Penida island. The spot is marked by a hilly arch-like rock formation, which is the distinguishable landmark of the area. Set over the crashing waves of the open Indian Ocean.",
                    "btnLabel": "View details",
                    "url": "/destinations/broken-beach"
                },
                {
                    "name": "Kelingking Beach",
                    "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/klingking-beach-mid-sq.jpg?alt=media&token=fe9953a8-95d2-4c70-97c8-ae8df6816bdb",
                    "desc": "Kelingking Beach is a hidden beach in the village of Bunga Mekar, on the southwestern coast of Nusa Penida island. You can enjoy one of the most breath-taking views over the hills and small strip of white sand from atop a hill of the same name. This unique formation is reminiscent of a Tyrannosaurus Rex head, hence its nickname ‘T-Rex Bay’.",
                    "btnLabel": "View details",
                    "url": "/destinations/kelingking-beach"
                },
                {
                    "name": "Angel Billabong",
                    "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/angel-billabong-mid-sq.jpg?alt=media&token=ef70e878-0de2-4b0c-bed8-513d3bc21758",
                    "desc": "a spectacular rock formation on Nusa Penida island's southwestern cliff edges. It's nearby Pasih Uug (Broken Beach), another popular site with unique limestone formations. The naturally formed rock lagoon offers a scenic seascape. You can descend into its crystal-clear rock pools for a swim or a soak",
                    "btnLabel": "View details",
                    "url": "/destinations/angel-billabong"
                },
                {
                    "name": "Broken Beach",
                    "img": "https://firebasestorage.googleapis.com/v0/b/nuxt-web-template.appspot.com/o/broken-beach-mid-sq.jpg?alt=media&token=9769ac16-f00a-494d-afb1-709e96327d6e",
                    "desc": "Broken Beach also popular as Pasih Uug is a scenic coastal formation on the southwestern edge of Nusa Penida island. The spot is marked by a hilly arch-like rock formation, which is the distinguishable landmark of the area. Set over the crashing waves of the open Indian Ocean.",
                    "btnLabel": "View details",
                    "url": "/destinations/broken-beach"
                }
            ]
        }
    ],
    "extras": [
        {
            "name": "Review",
            "googleReview": {
                "placeID": "ChIJ0xkTTRlx0i0Re3sZsgY3Olw"
            }
        }
    ]
}