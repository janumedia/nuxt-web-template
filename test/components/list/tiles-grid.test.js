import { shallowMount, mount, RouterLinkStub } from '@vue/test-utils'
import NuxtLink from '@/.nuxt/components/nuxt-link'
import TilesGrid from '@/components/list/tiles-grid'
import LinkedImage from '@/components/core/LinkedImage'
import Button from '@/components/core/Button'
import data from '../../fakeData/tiles-grid.data'

global.process.browser = true;
jest.useFakeTimers();

TilesGrid.components = TilesGrid.components || {};
TilesGrid.components.NuxtLink = NuxtLink;
LinkedImage.components = LinkedImage.components || {}
LinkedImage.components.NuxtLink = NuxtLink
Button.components = Button.components || {}
Button.components.NuxtLink = NuxtLink

const createWrapper = (propsData, mountType=shallowMount) => {
    return mountType(TilesGrid, {
        attachToDocument: true,
        propsData,
        stubs: {
            RouterLink: RouterLinkStub
        }
    })
}

describe("tiles-grid.vue", () => {
    let wrapper, items = data.tilesGrid;;
    beforeEach(() => {
        Object.defineProperty(Image.prototype, 'complete',  {
            value: true
        });
    });

    afterEach(() => {
        wrapper.destroy();

        Object.defineProperty(Image.prototype, 'complete',  {
            value: false
        });

        jest.clearAllTimers();
    });

    it('should render tiles-grid 2 columns', () => {
        wrapper = createWrapper({
            list: items.filter((item, index) => index < 3)
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render tiles-grid 2 colummns with thumbnailMode', () => {
        wrapper = createWrapper({
            list: items.filter((item, index) => index < 2),
            thumbnailMode: true
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render tiles-grid  3 columns', () => {
        wrapper = createWrapper({
            list: items.filter((item, index) => index < 3),
            numColumns: 3
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should firing imageReady', () => {
        wrapper = createWrapper({
            list: items.filter((item, index) => index < 3)
        }, mount);

        let imageReady = wrapper.vm.imageReady;
        let spy = jest.fn((e, index) => {
            imageReady(e, index);
            jest.runOnlyPendingTimers();
        })
        wrapper.setMethods({imageReady: spy});
        jest.runOnlyPendingTimers();
        expect(spy).toHaveBeenCalled();
    });
})