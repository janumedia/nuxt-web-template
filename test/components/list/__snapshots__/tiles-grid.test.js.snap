// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tiles-grid.vue should render tiles-grid  3 columns 1`] = `
<div class="tiles-container">
  <ul class="tiles tiles--col3">
    <li class="tile last-2nd" style="height: 0px;">
      <linkedimage-stub title="Name A" url="/link/to/a" src="imageA.jpg"></linkedimage-stub>
      <div class="tile__info">
        <h3>Name A</h3>
        <div>Description A</div>
        <div class="button-group" style="position: absolute; bottom: 14px; left: 0px; right: 0px;">
          <!---->
          <button-stub url="/link/to/a" label="View details"></button-stub>
          <br>
          <br>
        </div>
      </div>
    </li>
    <li class="tile" style="height: 0px;">
      <linkedimage-stub title="Name B" url="/link/to/b" src="imageB.jpg"></linkedimage-stub>
      <div class="tile__info">
        <h3>Name B</h3>
        <div>Description B</div>
        <div class="button-group" style="position: absolute; bottom: 14px; left: 0px; right: 0px;">
          <!---->
          <button-stub url="/link/to/b" label="View details"></button-stub>
          <br>
          <br>
        </div>
      </div>
    </li>
    <li class="tile" style="height: 0px;">
      <linkedimage-stub title="Name C" url="/link/to/c" src="imageC.jpg"></linkedimage-stub>
      <div class="tile__info">
        <h3>Name C</h3>
        <div>Description C</div>
        <div class="button-group" style="position: absolute; bottom: 14px; left: 0px; right: 0px;">
          <!---->
          <button-stub url="/link/to/c" label="View details"></button-stub>
          <br>
          <br>
        </div>
      </div>
    </li>
  </ul>
  <div class="clearfix"></div>
</div>
`;

exports[`tiles-grid.vue should render tiles-grid 2 colummns with thumbnailMode 1`] = `
<div class="tiles-container">
  <ul class="tiles tiles--col2">
    <li class="tile tile--thumbnail-mode last-2nd">
      <linkedimage-stub thumbnailmode="true" title="Name A" url="/link/to/a" src="imageA.jpg"></linkedimage-stub>
      <!---->
    </li>
    <li class="tile tile--thumbnail-mode">
      <linkedimage-stub thumbnailmode="true" title="Name B" url="/link/to/b" src="imageB.jpg"></linkedimage-stub>
      <!---->
    </li>
  </ul>
  <div class="clearfix"></div>
</div>
`;

exports[`tiles-grid.vue should render tiles-grid 2 columns 1`] = `
<div class="tiles-container">
  <ul class="tiles tiles--col2">
    <li class="tile" style="height: 0px;">
      <linkedimage-stub title="Name A" url="/link/to/a" src="imageA.jpg"></linkedimage-stub>
      <div class="tile__info">
        <h3>Name A</h3>
        <div>Description A</div>
        <div class="button-group" style="position: absolute; bottom: 14px; left: 0px; right: 0px;">
          <!---->
          <button-stub url="/link/to/a" label="View details"></button-stub>
          <br>
          <br>
        </div>
      </div>
    </li>
    <li class="tile" style="height: 0px;">
      <linkedimage-stub title="Name B" url="/link/to/b" src="imageB.jpg"></linkedimage-stub>
      <div class="tile__info">
        <h3>Name B</h3>
        <div>Description B</div>
        <div class="button-group" style="position: absolute; bottom: 14px; left: 0px; right: 0px;">
          <!---->
          <button-stub url="/link/to/b" label="View details"></button-stub>
          <br>
          <br>
        </div>
      </div>
    </li>
    <li class="tile last-center" style="height: 0px;">
      <linkedimage-stub title="Name C" url="/link/to/c" src="imageC.jpg"></linkedimage-stub>
      <div class="tile__info">
        <h3>Name C</h3>
        <div>Description C</div>
        <div class="button-group" style="position: absolute; bottom: 14px; left: 0px; right: 0px;">
          <!---->
          <button-stub url="/link/to/c" label="View details"></button-stub>
          <br>
          <br>
        </div>
      </div>
    </li>
  </ul>
  <div class="clearfix"></div>
</div>
`;
